Sources/URLSessionDecodable/URLSessionDecodable.swift

```diff
@@ -52,6 +51,28 @@ extension URLSession {
     }
     //swiftlint:enable function_parameter_count
 
+    /// Retrieves the contents of the specified URL and returns decoded response.
+    ///
+    /// The response data will be decoded to `T` type when `statusCode` is in range `200..<300`.
+    @available(iOS 13.0.0, tvOS 13.0.0, macOS 10.15.0, watchOS 8.0, *)
+    public func decodable<T: Decodable>(
+        with url: URL,
+        method: HTTPMethod,
+        parameters: ParametersEncoding?,
+        headers: HTTPHeaders?,
+        decoder: DataDecoder
+    ) async throws -> T {
+        let request = self.request(with: url, method: method, parameters: parameters, headers: headers)
+        let (data, response) = try await data(for: request)
```

Comment:
`URLSessionDecodableError` has a wrapper for URLSession errors but since the thrown error is not strongly typed I don't think we need that anymore